test/e2e/dataplane/service_discovery.go

```diff
@@ -42,4 +50,69 @@ func RunServiceDiscoveryTest(f *framework.Framework) {
 
 	Expect(err).NotTo(HaveOccurred())
 	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
+
+	kubeDnsService, _ := f.ClusterClients[framework.ClusterB].CoreV1().Services("kube-system").Get("kube-dns", metav1.GetOptions{})
+	kubeDnsServiceIP := kubeDnsService.Spec.ClusterIP
+
+	cmd = []string{"dig", "@" + kubeDnsServiceIP, "nginx-demo." + f.Namespace + ".svc.cluster2.local", "+short"}
+	stdout, _, err = f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).To(ContainSubstring(nginxServiceClusterCIP))
+	f.DeleteService(framework.ClusterC, "nginx-demo")
+	f.AwaitForMcsCrdDelete(framework.ClusterB, "nginx-demo")
+	stdout, _, err = f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).NotTo(ContainSubstring(nginxServiceClusterCIP))
+}
+
+func RunServiceDiscoveryLocalTest(f *framework.Framework) {
+	clusterBName := framework.TestContext.KubeContexts[framework.ClusterB]
+	clusterCName := framework.TestContext.KubeContexts[framework.ClusterC]
+
+	By(fmt.Sprintf("Creating an Nginx Deployment on on %q", clusterCName))
+	f.NewNginxDeployment(framework.ClusterB)
+
+	By(fmt.Sprintf("Creating a Nginx Service on %q", clusterCName))
+	nginxServiceClusterB := f.NewNginxService(framework.ClusterB)
+	nginxServiceClusterBIP := nginxServiceClusterB.Spec.ClusterIP
+
+	By(fmt.Sprintf("Creating an Nginx Deployment on on %q", clusterCName))
+	f.NewNginxDeployment(framework.ClusterC)
+
+	By(fmt.Sprintf("Creating a Nginx Service on %q", clusterCName))
+	_ = f.NewNginxService(framework.ClusterC)
+
+	By(fmt.Sprintf("Creating a Netshoot Deployment on %q", clusterBName))
+	netshootPodList := f.NewNetShootDeployment(framework.ClusterB)
+	kubeDnsService, _ := f.ClusterClients[framework.ClusterB].CoreV1().Services("kube-system").Get("kube-dns", metav1.GetOptions{})
+	kubeDnsServiceIP := kubeDnsService.Spec.ClusterIP
+
+	cmd := []string{"dig", "@" + kubeDnsServiceIP, "nginx-demo." + f.Namespace + ".svc.cluster2.local", "+short"}
+	stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).To(ContainSubstring(nginxServiceClusterBIP))
```

Comment:
I think it would be useful to extend this test scenario to delete the service on ClusterB and verify it's still discoverable via ClusterC.

Reply:
Done.